src/fondant/pipeline/runner.py

```diff
@@ -43,10 +44,17 @@ def _run(self, input_spec: str, *args, **kwargs):
         print("Starting pipeline run...")
 
         # copy the current environment with the DOCKER_DEFAULT_PLATFORM argument
-        subprocess.call(  # nosec
+        output = subprocess.run(  # nosec
             cmd,
             env=dict(os.environ, DOCKER_DEFAULT_PLATFORM="linux/amd64"),
+            capture_output=True,
+            encoding="utf8",
         )
+
+        if output.returncode != 0:
+            msg = f"Command failed with error: '{output.stderr}'"
```

Comment:
Does this properly show a full traceback with line numbers etc ?

Reply:
Just tested this and it does not seem to work well when working with docker-compose:

* using subprocess to call a python script with an error results in return_code 1 and proper error handling 
* using subprocess to call a docker compose that calls docker compose that launches a python script with an error results in return_code 0. However, `output.stderr` still has some content in it that is a proper traceback to the the error 

I'v added an example for testing 
https://github.com/ml6team/fondant/blob/handle-docker-compose-errors/example_subprocess/launch_subprocess.py

We could alternatively check if `stderr` is an empty string but not sure if that's the best method since I am not sure if warnings or other content gets streamed to it. 

Any suggestions?

Reply:
Fixed now by adding `--abort-on-container-exit` flag to docker compose